### PR TITLE
Add lambda friendly addCallback.

### PR DESCRIPTION
**Pull request by alanw**

### CODE REVIEW

**Comment by rschildmeijer** [comment]
Looks good.

 My only concern is that we allow null for the consumers (for one or for both).

File: README.md [hunk 1]
```diff
@@ -158,8 +158,15 @@ final ListenableFuture<A> firstSuccessful = FuturesExtra.select(futures);
 
 #### Success/Failure callbacks
 
-If you are only interested in either successful or failed results of a future,
-you can use these callbacks:
+You can attach callbacks that are run depending on the results of a future:
+```java
+final ListenableFuture<A> future = getFuture();
+FuturesExtra.addCallback(future, a -> System.out.println(a), e -> e.printStackTrace());
```
Comment (spkrka) [inline README.md]: this example could also use method references: System.out::println and Throwable::printStackTrace

File: src/main/java/com/spotify/futures/FuturesExtra.java [hunk 2]
```diff
@@ -172,47 +172,56 @@ public void onFailure(Throwable t) {
...
+   * @param failure a consumer, to be called with the result of the failed future.
    */
-  public static <T> void addSuccessCallback(
-      ListenableFuture<T> future,
-      final Consumer<? super T> consumer) {
+  public static <T> void addCallback(
+          final ListenableFuture<T> future,
+          final Consumer<? super T> success,
+          final Consumer<Throwable> failure) {
     Futures.addCallback(future, new FutureCallback<T>() {
```
Comment (spkrka) [inline src/main/java/com/spotify/futures/FuturesExtra.java]: verify that both are not null? or at least add a quick return if both are null

File: src/main/java/com/spotify/futures/FuturesExtra.java [hunk 3]
```diff
@@ -172,47 +172,59 @@ public void onFailure(Throwable t) {
...
-   * A lambda-friendly way to attach a callback to a {@link ListenableFuture}. The callback will
-   * only be run if the future completes successfully.
-   * @param future a ListenableFuture to attach the callback to.
-   * @param consumer a consumer, to be called with the result of the successful future.
+   * A lambda-friendly way to attach callbacks to a {@link ListenableFuture}. The success
+   * callback will only be run if the future completes successfully, and failure will
+   * only be run if the future fails.
+   * @param future a ListenableFuture to attach the callbacks to.
+   * @param success a consumer, to be called with the result of the successful future.
+   * @param failure a consumer, to be called with the result of the failed future.
```
Comment (spkrka) [inline src/main/java/com/spotify/futures/FuturesExtra.java]: document the nullability-rules of the parameters (both may not be null)

File: src/main/java/com/spotify/futures/FuturesExtra.java [hunk 4]
```diff
@@ -172,47 +172,59 @@ public void onFailure(Throwable t) {
...
    */
-  public static <T> void addSuccessCallback(
-      ListenableFuture<T> future,
-      final Consumer<? super T> consumer) {
+  public static <T> void addCallback(
+          final ListenableFuture<T> future,
+          final Consumer<? super T> success,
+          final Consumer<Throwable> failure) {
+    if (success == null && failure == null) {
+      throw new NullPointerException();
```
Comment (spkrka) [inline src/main/java/com/spotify/futures/FuturesExtra.java]: add unit test for passing in both null

File: src/main/java/com/spotify/futures/FuturesExtra.java [hunk 5]
```diff
@@ -172,47 +172,59 @@ public void onFailure(Throwable t) {
...
+    }
     Futures.addCallback(future, new FutureCallback<T>() {
       @Override
-      public void onSuccess(T result) {
-        consumer.accept(result);
+      public void onSuccess(final T result) {
+        if (success != null) {
+          success.accept(result);
```
Comment (spkrka) [inline src/main/java/com/spotify/futures/FuturesExtra.java]: add unit test to explicitly test for when success == null

File: src/main/java/com/spotify/futures/FuturesExtra.java [hunk 6]
```diff
@@ -172,47 +172,59 @@ public void onFailure(Throwable t) {
...
+        }
       }
 
       @Override
-      public void onFailure(Throwable t) {
-        // ignore
+      public void onFailure(final Throwable throwable) {
+        if (failure != null) {
+          failure.accept(throwable);
```
Comment (spkrka) [inline src/main/java/com/spotify/futures/FuturesExtra.java]: add unit test to explicitly test for when failure == null

File: src/main/java/com/spotify/futures/FuturesExtra.java [hunk 7]
```diff
@@ -172,47 +172,60 @@ public void onFailure(Throwable t) {
...
-   * only be run if the future completes successfully.
-   * @param future a ListenableFuture to attach the callback to.
-   * @param consumer a consumer, to be called with the result of the successful future.
+   * A lambda-friendly way to attach callbacks to a {@link ListenableFuture}. The success
+   * callback will only be run if the future completes successfully, and failure will
+   * only be run if the future fails.
+   * @param future a ListenableFuture to attach the callbacks to.
+   * @param success a consumer, to be called with the result of the successful future.
+   * @param failure a consumer, to be called with the result of the failed future.
+   * @throws NullPointerException if the {@param success} and {@param failure} are null
```
Comment (spkrka) [inline src/main/java/com/spotify/futures/FuturesExtra.java]: this is slightly unclear (not obvious that both have to be null for it to throw) but i dont want to reject the change based on that.
